bitcoin-core-sv2/examples/logger.rs

```diff
@@ -0,0 +1,164 @@
+//! A simple example of how to use `BitcoinCoreSv2`.
+//!
+//! We connect to the Bitcoin Core UNIX socket, and log the received Sv2 Template Distribution
+//! Protocol messages.
+//!
+//! Every 10s, we send a new `CoinbaseOutputConstraints` message to the `BitcoinCoreSv2` instance.
+//!
+//! `BitcoinCoreSv2` will not start distributing new templates until it receives the first
+//! `CoinbaseOutputConstraints` message.
+
+use bitcoin_core_sv2::BitcoinCoreSv2;
+use std::path::Path;
+
+use async_channel::unbounded;
+use stratum_core::{
+    parsers_sv2::TemplateDistribution,
+    template_distribution_sv2::{CoinbaseOutputConstraints, RequestTransactionData},
+};
+use tokio_util::sync::CancellationToken;
+use tracing::info;
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt::init();
+
+    // the user must provide the path to the Bitcoin Core UNIX socket
```

Comment:
Would be a nice followup to find the default socket path based on just the network. Though a custom path should always be possible.

https://en.bitcoin.it/wiki/Data_directory

One caveat, especially for CI: different operating systems have a fairly short maximum length for unix socket paths. See e.g. https://github.com/bitcoin/bitcoin/blob/115d298a9fa31ffacd980381e2f459c9440d8833/test/functional/test_framework/test_node.py#L139-L148